<launch>
  <!--launch-prefix="gdb -ex run - -args" -->

  <node name="simulate_nbv_cycle" type="simulate_nbv_cycle" pkg="nbv_3d_cnn" output="screen" clear_params="true" >
    <param name="image_file_name" type="string" value="$(find nbv_3d_cnn)/data/test_image/179_environment_topleft.png" />
    <!--<param name="image_file_name" type="string" value="$(find nbv_3d_cnn)/data/test_image/austin1.png" />-->
    <param name="debug_output_folder" type="string" value="$(find nbv_3d_cnn)/data/simulate_nbv_cycle/" />

    <param name="random_seed" type="int" value="0" />

    <param name="max_iterations" type="int" value="100" />

    <param name="predict_action_name" type="string" value="/nbv_3d_cnn_predict/predict" />
    <param name="predict_quat_action_name" type="string" value="/nbv_3d_cnn_predict_quat/predict" />
    <param name="predict_autocomplete_action_name" type="string" value="/nbv_3d_cnn_predict_autocomplete/predict" />
    <param name="predict_circular_action_name" type="string" value="/nbv_3d_cnn_predict_circular/predict" />
    <param name="predict_flat_action_name" type="string" value="/nbv_3d_cnn_predict_flat/predict" />

    <param name="save_images" type="bool" value="false" />

    <!--
    <param name="nbv_algorithm" type="string" value="Random" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/random_nbv_log.txt" />
    -->
    
    <!--
    <param name="nbv_algorithm" type="string" value="CNNDirectDirectional" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/cnn_ov_nbv_log.txt" />
    -->

    
    <param name="nbv_algorithm" type="string" value="CNNQuat" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/cnn_sa_nbv_log.txt" />
    

    <!--
    <param name="nbv_algorithm" type="string" value="InformationGain" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/ig_nbv_log.txt" />
    -->

    <!--
    <param name="nbv_algorithm" type="string" value="CNNFlat" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/cnn_ov_flat_nbv_log.txt" />
    -->

    <!--
    <param name="nbv_algorithm" type="string" value="InformationGainProb" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/ig_nbv_log_prob.txt" />
    -->

    <!--
    <param name="nbv_algorithm" type="string" value="AutocompleteIGain" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/ig_nbv_log_autocomplete.txt" />
    -->
 
    <!--
    <param name="nbv_algorithm" type="string" value="OmniscientGain" />
    <param name="log_file" type="string" value="$(find nbv_3d_cnn)/data/logs/omni_nbv_log.txt" />
    -->

    <param name="igain_min_range" type="double" value="15.0" />

    <param name="sensor_focal_length" type="double" value="450" />

    <param name="sensor_range_voxels" type="double" value="128" />

    <param name="view_cube_resolution" type="int" value="16" />

    <param name="a_priori_occupied_prob" type="double" value="0.15" />
  </node>

  <node name="nbv_3d_cnn_predict" type="nbv_3d_cnn_predict.py" pkg="nbv_3d_cnn" output="screen" clear_params="true">
    <param name="checkpoint_file" type="string" value="$(find nbv_3d_cnn)data/trained_models/_final.chkpt" />

    <param name="sub_image_expand_pow" type="int" value="2" />

    <param name="model_type" type="string" value="" />
  </node>

  <node name="nbv_3d_cnn_predict_circular" type="nbv_3d_cnn_predict.py" pkg="nbv_3d_cnn" output="screen" clear_params="true">
    <param name="checkpoint_file" type="string" value="$(find nbv_3d_cnn)data/trained_models/circular_final.chkpt" />

    <param name="sub_image_expand_pow" type="int" value="2" />

    <param name="model_type" type="string" value="circular" />
  </node>

  <node name="nbv_3d_cnn_predict_flat" type="nbv_3d_cnn_predict.py" pkg="nbv_3d_cnn" output="screen" clear_params="true">
    <param name="checkpoint_file" type="string" value="$(find nbv_3d_cnn)data/trained_models/flat_final.chkpt" />

    <param name="sub_image_expand_pow" type="int" value="2" />

    <param name="model_type" type="string" value="flat" />
  </node>

  <node name="nbv_3d_cnn_predict_quat" type="nbv_3d_cnn_predict.py" pkg="nbv_3d_cnn" output="screen" clear_params="true">
    <param name="checkpoint_file" type="string" value="$(find nbv_3d_cnn)data/trained_models/scoreangle_final.chkpt" />

    <param name="sub_image_expand_pow" type="int" value="0" />
    
    <param name="model_type" type="string" value="quat" />
  </node>

  <node name="nbv_3d_cnn_predict_autocomplete" type="nbv_3d_cnn_predict.py" pkg="nbv_3d_cnn" output="screen" clear_params="true">
    <param name="checkpoint_file" type="string" value="$(find nbv_3d_cnn)data/trained_models/autocomplete_final.chkpt" />

    <param name="sub_image_expand_pow" type="int" value="0" />
    
    <param name="model_type" type="string" value="autocomplete" />
  </node>
</launch>
